corehq/apps/domain/decorators.py

```diff
@@ -147,6 +147,36 @@ def wrapper(request, *args, **kwargs):
     return real_decorator
 
 
+def _superuser_auth_decorator(challenge_fn):
+    def _outer(fn):
+        @wraps(fn)
+        def safe_fn(request, *args, **kwargs):
+            if args:
+                domain_name, domain = load_domain(request, args[0])
+                if domain:
+                    raise Exception(
+                        "This decorator doesn't do domain-specific validation. Please use decorators "
+                        "based on _login_or_challenge for domain-specific auth and validation"
+                    )
+
+            @check_lockout
+            @challenge_fn
+            @require_superuser_or_developer
```

Comment:
is `or_developer` right? i think it's consistent with the prior code, but if that's the intention it might be better to include it in the name of the function